### PR TITLE
Support the latest version of pkg:analyzer

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
@munificent – before we release 2.0.2, we should allow the latest analyzer, too!

**Comment by kevmoo** [comment]
@munificent – note: this PR goes against your `bump` branch – FYI

File: pubspec.yaml [hunk 1]
```diff
@@ -31,3 +31,7 @@ dev_dependencies:
 executables:
   dartfmt: format
   dartformat: format # Allow the old name for compatibility.
+
+# For CI - remove before commiting!
+dependency_overrides:
+  analyzer: ^2.0.0
```
Comment (munificent) [inline pubspec.yaml:37]: What's this about?

Comment (kevmoo) [reply pubspec.yaml:37]: Forcing CI to run w/ v2.0.0 (it's blocked by pkg:test at the moment) to make sure there are no breaks.

I can nix this now, or you can merge this into your PR. Let CI run w/ it, then remove it.

Just to make sure we have coverage of the new API

Comment (munificent) [reply pubspec.yaml:37]: SGTM. I'll merge this into the "bump" branch. Once CI looks good there, I'll remove this and then merge to master and publish.

**Pull request merged**